tools/methyldackel/MethylDackel.xml

```diff
@@ -104,301 +110,487 @@
+            </conditional>
+            <section name="advanced_options">       
+                <param name="min_mapq" value="2" />
+            </section>
+            <output name="outFileExtractCpGLogit" ftype="bedgraph">
+                <assert_contents>
+                     <has_size value="845"/>
+                </assert_contents>
+            </output> 
         </test>
-        <test>
-            <param name="task" value="extract" />
-            <param name="min_mapq" value="2" />
+        <test expect_num_outputs="1">
             <param name="reference_source_selector" value="history" />
             <param name="ref_file" value="cg100.fa" ftype="fasta" />
             <param name="input_sortedAlignBAM" value="cg_aln.bam" ftype="bam"/>
-            <param name="options" value="yes"/>
-            <param name="methylKit" value="true" />
-            <output name="outFileExtractCpGMethylKit" file="test_5.methylKit" ftype="text" compare="diff"/>
+            <conditional name="main_task">
+                <param name="task" value="extract" />
+                <param name="extract_outputs" value="--methylKit" />
+            </conditional>
+            <section name="advanced_options">
+                <param name="min_mapq" value="2" />
+            </section>
+            <output name="outFileExtractCpGMethylKit" ftype="txt">
+                <assert_contents>
+                     <has_size value="1763"/>
+                </assert_contents>
+            </output> 
+
         </test>
-        <test>
-            <param name="task" value="mbias" />
-            <param name="min_mapq" value="2" />
-            <param name="options" value="yes"/>
-            <param name="tsv" value="--txt"/>
+        
+        <test expect_num_outputs="5">
             <param name="reference_source_selector" value="history" />
             <param name="ref_file" value="cg100.fa" ftype="fasta" />
             <param name="input_sortedAlignBAM" value="cg_aln.bam" ftype="bam"/>
-            <output name="outFileMbiasTSV" file="test7_output.tsv" ftype="tabular" compare="diff"/>
+            <conditional name="main_task">
+                <param name="task" value="mbias" />
+                <param name="tsv" value="--txt"/>
+            </conditional>
+            <section name="advanced_options">
+                <param name="min_mapq" value="2" />
+            </section>
+            <output name="outFileMbiasTSV" ftype="tabular">
+                <assert_contents>
+                     <has_size value="1213"/>
+                </assert_contents>
+            </output> 
+            <output name="outFileMbiasCpGOT" ftype="svg">
+                 <assert_contents>
+                     <has_size value="14614"/>
+                </assert_contents>
+            </output> 
+            <output name="outFileMbiasCpGOB"  ftype="svg">
+                 <assert_contents>
+                     <has_size value="0"/>
```

Comment:
https://github.com/bgruening/galaxytools/pull/1130/files#r645554144

Those files are empty, but in the previous release the tests didn't check them correctly.